edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.AuthAgent/AuthAgentListener.cs

```diff
@@ -0,0 +1,532 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.AuthAgent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class AuthAgentListener : IProtocolHead
+    {
+        private readonly IAuthenticator authenticator;
+        private readonly IUsernameParser usernameParser;
+        private readonly IClientCredentialsFactory clientCredentialsFactory;
+        private readonly string listeningAddress;
+
+        private HttpListener listener;
+
+        private Task listenerTask;
+        private bool exitListenerLoop;
+
+        private const int maxInBufferSize = 32 * 1024;
```

Comment:
comment here reasoning as well? I dislike seemingly arbitrary consts with no explination.

Reply:
added an explanation how the buffer size was picked